precalculus/exercises/outcomes/EQ/EQ2/generator.sage

```diff
@@ -1,10 +1,153 @@
 class Generator(BaseGenerator):
     def data(self):
-      return {
-        "eq": 3,
-        "ineq": 4,
-        "eqansinterval": 5,
-        "eqansgraph": 6,
-        "ineqansinterval": 7,
-        "ineqansgraph": 8
-      } 
\ No newline at end of file
+        scenario = choice([
+            "distance_toward",
+            "distance_apart",
+            "mixture_value",
+            "mixture_percent",
+        ])
+        scenario_alt = choice([
+            "scenarioA",
+            "scenarioB",
+            "scenarioC",
+            "scenarioD",
+        ])
+        name = choice([
+            "Anne-Fatima",
+            "Boram",
+            "Consolata",
+            "Dalisay",
+            "Edgar",
+            "Fionnuala",
+            "Gurlez",
+            "Hisako",
+            "Inga",
+            "Julia",
+        ])
+
+        data = {}
+
+        if scenario == "distance_toward" or scenario == "distance_apart":
+            d_unit = choice(["mi","km"])
+            vehicles = choice(["trains", "cars", "trucks"])
+            directions = choice([
+                ["north", "south"],
+                ["east", "west"],
+                ["northeast", "southwest"],
+                ["northwest", "southeast"],
+            ])
+            faster_slower = choice(["faster", "slower"])
+            shuffle(directions)
+            mults = sample(range(4,11),2)
+            rates = [20*m for m in mults]
+            time = choice([3,5,7,9,11])/4  # hours
+            dist = sum(rates)*time
+            data = {
+                "dist": dist,
+                "rate1": rates[0],
+                "rate2": rates[1],
+                "rate_diff": abs(rates[0]-rates[1]),
+                "time": time,
+                "d_unit": d_unit,
+                "vehicles": vehicles,
+                "direction1": directions[0],
+                "direction2": directions[1],    
+                "faster_slower": faster_slower,   
+            }
+        
+        if scenario == "mixture_value":
+            part1, part2, percentage = sample([10,20,30,40,60,70,80,90], 3)
+            mixpart = part1*percentage/100 + part2*(100-percentage)/100
+            totalamount = choice(range(10,100,10))
+            amount1 = totalamount*percentage/100
+            amount2 = totalamount*(100-percentage)/100
+
+            if scenario_alt[-1] in ["A", "B"]:
+                weight = choice(["lb", "kg"])
```

Comment:
Are we confident all students will be comfortable with the "lb" abbreviation, or would we be better of just saying "pounds" here?

Reply:
best to keep weights all abbreviated or not; I think since there's no dimensional analysis happening here it's fine as-is until we hear otherwise.